controllers/drplacementcontrol_controller.go

```diff
@@ -401,6 +434,55 @@ func (r *DRPlacementControlReconciler) clonePlacementRule(ctx context.Context,
 	return clonedPlRule, nil
 }
 
+func (r *DRPlacementControlReconciler) getSchedule(ctx context.Context,
+	drpc *rmn.DRPlacementControl) (*string, error) {
+	r.Log.Info("Getting schedule from DRPolicy")
+
+	clPeersRef := drpc.Spec.DRPolicyRef
+	clusterPeers := &rmn.DRPolicy{}
+	schedule := ""
+
+	err := r.Client.Get(ctx,
+		types.NamespacedName{Name: clPeersRef.Name}, clusterPeers)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster peers using %s/%s. Error (%w)",
+			clPeersRef.Name, clPeersRef.Namespace, err)
+	}
+
+	if clusterPeers.Spec.Schedule == nil {
+		return nil, fmt.Errorf("invalid DRPolicy configuration. Schedule empty (%s)", clusterPeers.Name)
+	}
+
+	parser := cron.NewParser(cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow | cron.Descriptor)
+	if _, err = parser.Parse(*clusterPeers.Spec.Schedule); err != nil {
+		return nil, fmt.Errorf("error parsing schedule %s", *clusterPeers.Spec.Schedule)
+	}
+
+	schedule = *clusterPeers.Spec.Schedule
+
+	return &schedule, nil
+}
+
+func (r *DRPlacementControlReconciler) getReplicationClassSelector(ctx context.Context,
+	drpc *rmn.DRPlacementControl) (*metav1.LabelSelector, error) {
+	r.Log.Info("Getting schedule from DRPolicy")
+
+	clPeersRef := drpc.Spec.DRPolicyRef
+	clusterPeers := &rmn.DRPolicy{}
+	replicationClassSelector := &metav1.LabelSelector{}
+
+	err := r.Client.Get(ctx,
+		types.NamespacedName{Name: clPeersRef.Name}, clusterPeers)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster peers using %s/%s. Error (%w)",
+			clPeersRef.Name, clPeersRef.Namespace, err)
+	}
+
+	*replicationClassSelector = clusterPeers.Spec.ReplicationClassSelector
+
+	return replicationClassSelector, nil
```

Comment:
Why don't you just return the copy of `clusterPeers.Spec.ReplicationClassSelector` instead of creating an instance in line 473 and then assignment in line 481.